controllers/Index.php

```diff
@@ -112,52 +111,36 @@ public function password() {
     }
 
     /**
-     * check and show login/logout
-     * html
-     *
-     * @return void
-     */
-    private function authentication() {
-        // logout
-        if (isset($_GET['logout'])) {
-            \F3::get('auth')->logout();
-            \F3::reroute($this->view->base);
-        }
-
-        // login
-        $loginRequired = \F3::get('public') != 1 && \F3::get('auth')->isLoggedin() !== true;
-        $showLoginForm = isset($_GET['login']) || $loginRequired;
-        if ($showLoginForm) {
-            // authenticate?
-            if (count($_POST) > 0) {
-                if (!isset($_POST['username'])) {
-                    $this->view->error = 'no username given';
-                } elseif (!isset($_POST['password'])) {
-                    $this->view->error = 'no password given';
-                } elseif (!\F3::get('auth')->login($_POST['username'], $_POST['password'])) {
-                    $this->view->error = 'invalid username/password';
-                }
-            }
-
-            // show login
-            if (count($_POST) === 0 || isset($this->view->error)) {
-                die($this->view->render('templates/login.phtml'));
-            } else {
-                \F3::reroute($this->view->base);
-            }
-        }
-    }
-
-    /**
      * login for api json access
      * json
      *
      * @return void
      */
     public function login() {
         $view = new \helpers\View();
-        $username = isset($_REQUEST['username']) ? $_REQUEST['username'] : '';
-        $password = isset($_REQUEST['password']) ? $_REQUEST['password'] : '';
+
+        $error = null;
+
+        if (isset($_REQUEST['username'])) {
+            $username = $_REQUEST['username'];
+        } else {
+            $username = '';
+            $error = 'no username given';
+        }
+
+        if (isset($_REQUEST['password'])) {
+            $password = $_REQUEST['password'];
+        } else {
+            $password = '';
+            $error = 'no password given';
+        }
+
+        if ($error !== null) {
+            $view->jsonSuccess([
```

Comment:
It made not much sense but the login API was like that before so I did ont change et.